addon/test-waiter.ts

```diff
@@ -19,6 +19,7 @@ export default class TestWaiter<T = Token> implements ITestWaiter<T> {
   private isRegistered = false;
 
   items = new Map<T, ITestWaiterDebugInfo>();
+  endedItems = new Map<T, boolean>();
```

Comment:
Can we avoid needing the extra map here, by weakly associating the `TestWaiter` with the token?

Reply:
To be clear, Rob is suggesting we do

```ts
let trackedTokens = new WeakMap<Token, TestWaiter>();
```

to track the association of the token with a particular test waiter instance, _and_ have the ability to detect if a token is attempting to be used to invoke `endAsync` more than once.